temp/legacy-code/src/main/java/com/ivy/legacy/utils/AmountFormatting.kt

```diff
@@ -105,7 +105,22 @@ fun shortenAmount(amount: Double): String {
             formatShortenedNumber(amount / THOUSAND, "k")
         }
 
-        else -> amount.toString()
+        else -> {
+            val formattedAmount = amount.format(2) // Format with 2 decimal places
+            if (formattedAmount.contains(".")) {
```

Comment:
I'm not sure if this will work in all locales. Even if it works it's too complex. Another concern is that affects formatting globally for the app including for crypto.

Let's just in the widget code only:
```kotlin
DecimalFormat("###,###.##").format(balance)`
```

That feels like the safest and most straightforward change.

Reply:
Ok then i will do this then.